src/components/Dialog/Dialog.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { FC, useEffect } from 'react';
+
+// Define the props
+export interface DialogProps {
+  /** control the open/close of the dialog from props */
+  isOpen?: boolean;
+  /** listen to the close of the dialog by providing the onClose callback function */
+  onClose?: () => void; //e: React.MouseEvent
+  /** Developers can provide title, content, actions of the dialog component from children props of the component */
+  children?: React.ReactNode;
+}
+
+interface DialogTitleProps {
+  /** any elements */
+  children?: React.ReactNode;
+}
+
+interface DialogContentProps {
+  /** any elements */
+  children?: React.ReactNode;
+}
+
+interface DialogActionsProps {
+  /** any elements */
+  children?: React.ReactNode;
```

Comment:
It will be better to name it actionsPropsChildren to explicitly show this is the children prop for the action.